Core/Src/main.c

```diff
@@ -87,14 +89,38 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();       // SPI is initialized here
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  uint8_t inputData[3] = {0x01, 0x80, 0x00};
+  uint8_t outputData[3] = {0};
+  uint16_t dataSize = 3;
```

Comment:
This variable doesn't change right? Consider making it const and following the ALL_CAPS convention for constant variables.